crates/core_arch/src/arm/neon.rs

```diff
@@ -1136,12 +1388,526 @@ pub unsafe fn vtbx4_p8(a: poly8x8_t, b: poly8x8x4_t, c: uint8x8_t) -> poly8x8_t
     ))
 }
 
+macro_rules! arm_vget_lane {
+    ($name:ident, $from:ty, $to:ty, $lanes:literal) => {
+        #[inline]
+        #[target_feature(enable = "neon")]
+        #[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+        #[cfg_attr(test, assert_instr(umov))]
+        pub unsafe fn $name(v: $from, lane: i32) -> $to {
+            macro_rules! call {
+                ($imm5:expr) => {
+                    if ($imm5) < 0 || ($imm5) > $lanes {
+                        unreachable_unchecked()
+                    } else {
+                        simd_extract(v, $imm5)
+                    }
+                };
+            }
+            constify_imm5!(lane, call)
+        }
+    };
+}
+
+//  uint64_t vgetq_lane_u64 (uint64x2_t v, const int lane)
+arm_vget_lane!(vgetq_lane_u64, uint64x2_t, u64, 2);
+
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+#[rustc_args_required_const(1)]
+#[cfg_attr(test, assert_instr(umov, imm5 = 0))]
+pub unsafe fn vgetq_lane_u16(v: uint16x8_t, imm5: i32) -> u16 {
```

Comment:
Can this be an `u32`? (does stdarch-verify allow that?)

Reply:
will try!

Reply:
tried, stdarch-verify complains, sorry.

Reply:
From looking at the actual C type, which isn't `int`, but  `__constrange(0,Y) int`, `stdarch-verify` probably has a bug.

Reply:
cc @eddyb it appears that we might have to extend `#[rustc_args_required_const(1)]` with range information